Fluid/TemplateContext.cs

```diff
@@ -1,274 +1,280 @@
+        public int MaxSteps { get; set; } = TemplateOptions.Default.MaxSteps;
+
+        /// <summary>
+        /// Gets <see cref="StringComparer"/> used when comparing model names.
+        /// </summary>
+        public StringComparer ModelNamesComparer { get; private set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="CultureInfo"/> instance used to render locale values like dates and numbers.
+        /// </summary>
+        public CultureInfo CultureInfo { get; set; } = TemplateOptions.Default.CultureInfo;
+
+        /// <summary>
+        /// Gets or sets the value to returned by the "now" keyword.
+        /// </summary>
+        public Func<DateTimeOffset> Now { get; set; } = TemplateOptions.Default.Now;
+
+        /// <summary>
+        /// Gets or sets the local time zone used when parsing or creating dates without specific ones.
+        /// </summary>
+        public TimeZoneInfo TimeZone { get; set; } = TemplateOptions.Default.TimeZone;
+
+        /// <summary>
+        /// Increments the number of statements the current template is processing.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void IncrementSteps()
+        {
+            if (MaxSteps > 0 && _steps++ > MaxSteps)
+            {
+                ExceptionHelper.ThrowMaximumRecursionException();
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the current scope.
+        /// </summary>
+        internal Scope LocalScope { get; set; }
+
+        /// <summary>
+        /// Gets or sets the root scope.
+        /// </summary>
+        internal Scope RootScope { get; set; }
+
+        private Dictionary<string, object> _ambientValues;
+
+        /// <summary>
+        /// Used to define custom object on this instance to be used in filters and statements
+        /// but which are not available from the template.
+        /// </summary>
+        public Dictionary<string, object> AmbientValues => _ambientValues ??= new Dictionary<string, object>();
+
+        /// <summary>
+        /// Gets or sets a model object that is used to resolve properties in a template. This object is used if local and
+        /// global scopes are unsuccessful.
+        /// </summary>
+        public FluidValue Model { get; }
+
+        /// <summary>
+        /// Whether the direct properties of the Model can be accessed without being registered. Default is <code>true</code>.
+        /// </summary>
+        public bool AllowModelMembers { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets the delegate to execute when a Capture tag has been evaluated.
+        /// </summary>
+        public Func<string, string, ValueTask<string>> Captured { get; set; }
+
+        /// <summary>
+        /// Gets or sets the delegate to execute when a member has been assigned via capture or assign
+        /// </summary>
+        public Action<FluidValue> Assigned { get; set; }

```

Comment:
Like `Captured` this should return a `ValueTask`. But unlike `Captured`, and that's something that could be changed at some point, we should also provide the `TemplateContext` as an argument because sometimes you need to take the options into account to do something. I have had to add it in the most recent commit I did, and obsolete the previous methods. So let's not miss it here.

Reply:
So, should the delegate be `async`? (warning: async programming is a weak spot for me...)

Reply:
What it means is the delegate signature needs to return a `ValueTask` so instead of `Action<FluidValue>` it becomes a `Func<FluidValue, ValueTask>`. Then to add `TemplateContext` as the input, it becomes `Func<FluidValue, TemplateContext, ValueTask>`.

I will need to help you though for the invokation since there is an early non-async exit (`Awaited` method) and I don't expect you to know how to handle that (you can try still).

Reply:
...so, I wasn't planning on do anything with the return value. My goal was just to get variables "out" of the template. Do you want something to return and be assigned inside the context?

Reply:
Let me update your PR, you'll see. To make something "async" you just need to return "ValueTask<T>" or "Task<T>", you don't use it, the compiler/runtime will.